crates/stc_ts_file_analyzer/src/analyzer/expr/object.rs

```diff
@@ -39,6 +39,16 @@ impl Analyzer<'_, '_> {
             let mut known_keys = vec![];
             for prop in node.props.iter() {
                 ret = a.append_prop_or_spread_to_type(&mut known_keys, ret, prop, type_ann.as_deref())?;
+
+                if ret.is_unknown() {
+                    if let RPropOrSpread::Spread(..) = prop {
+                        return Err(ErrorKind::NonObjectInSpread {
```

Comment:
I think this needs to be `self.storage.report` because otherwise it won't catch 2 unknown spread elements ex:

```ts
function fn1(a: unknown, b: unknown) {
  let o1 = { ...a, ...b };
}
```

currently only `a` gets the error but TypeScript (from the IDE) seems to add the error to both `a` and `b`

Reply:
I think you are right

Reply:
I tried with `self.storage.report` and `self.storage.report_all` but I still got the same results. As in, neither of them would report the error on `...d` here, only on the first one

```ts
function f26(x: {}, y: unknown, z: any, d: unknown) {
  let o2 = { a: 42, ...y, ...d, ...x }; // unknown
  // only ...y is reported by stc
}
```

Reply:
My rough guess is that you used the same span, but I think it's fine to postpone it for another PR.